src/main/java/br/com/faateodoro/nossacasadocodigo/controller/dto/LivroDto.java

```diff
@@ -0,0 +1,87 @@
+package br.com.faateodoro.nossacasadocodigo.controller.dto;
+
+import br.com.faateodoro.nossacasadocodigo.modelo.Autor;
+import br.com.faateodoro.nossacasadocodigo.modelo.Categoria;
+import br.com.faateodoro.nossacasadocodigo.modelo.Livro;
+import br.com.faateodoro.nossacasadocodigo.repository.AutorRepository;
+import br.com.faateodoro.nossacasadocodigo.repository.CategoriaRepository;
+import br.com.faateodoro.nossacasadocodigo.validacao.exception.ValidationException;
+import org.hibernate.validator.constraints.Length;
+
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.Future;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.Optional;
+
+public class LivroDto {
+    @NotBlank(message = "{livro.titulo.obrigatorio}")
+    private String titulo;
+
+    @Length(max=500, message = "{livro.resumo.maximo}")
+    @NotBlank(message = "{livro.resumo.obrigatorio}")
+    private String resumo;
+
+    @NotBlank(message = "{livro.sumario.obrigatorio}")
+    private String sumario;
+
+    @DecimalMin(value="20.0", message = "{livro.preco.minimo}")
+    private BigDecimal preco;
+
+    @Min(value=100, message = "{livro.paginas.minimo}")
+    private Integer paginas;
+
+    @NotBlank(message = "{livro.isbn.obrigatorio}")
+    private String isbn;
+
+    @Future(message = "{livro.publicacao.futuro}")
+    private LocalDate dataPublicacao;
+
+    @NotBlank(message = "{livro.categoria.obrigatorio}")
+    private String nomeCategoria;
+
+    @NotBlank(message = "{libro.autor.obrigatorio}")
+    private String emailAutor;
+
+    public LivroDto(String titulo, String resumo, String sumario, BigDecimal preco,
+                    Integer paginas, String isbn, LocalDate dataPublicacao, String nomeCategoria, String emailAutor) {
+        this.titulo = titulo;
+        this.resumo = resumo;
+        this.sumario = sumario;
+        this.preco = preco;
+        this.paginas = paginas;
+        this.isbn = isbn;
+        this.dataPublicacao = dataPublicacao;
+        this.nomeCategoria = nomeCategoria;
+        this.emailAutor = emailAutor;
+    }
+
+    public Livro ToLivro(CategoriaRepository categoriaRepository, AutorRepository autorRepository) {
```

Comment:
O maiúsculo aqui tá fora da convenção do Java!

```suggestion
    public Livro toLivro(CategoriaRepository categoriaRepository, AutorRepository autorRepository) {
```

Reply:
Eita, codei C# recentemente e olha no que deu...